packages/server/src/restaurants/database/DishesSchema.ts

```diff
@@ -0,0 +1,185 @@
+
+enum DishTags {
+  Vegan = 'wegańska',
+  Vegetarian = 'wegetariańska',
+  GlutenFree = 'gluten free',
+  Spicy = 'ostre',
+  VerySpicy = 'bardzo ostre',
+}
+
+enum MealType {
+  Breakfast = 'śniadanie',
+  Lunch = 'lunch',
+  Dinner = 'obiad',
+  Supper = 'kolacja',
+}
+
+enum Allergens {
+  Gluten = 'gluten',
+  ShellFish = 'skorupiaki',
+  Eggs = 'jaja',
+  Fish = 'ryby',
+  Nuts = 'orzechy',
+  Soya = 'soja',
+  Milk = 'mleko',
+  Celery = 'seler',
+  Mustard = 'gorczyca',
+  SesameSeeds = 'nasiona sezamu',
+  SulphurDioxide = 'dwutlenek siarki',
+  Lupine = 'łubin',
+  Molluscs = 'mięczaki',
+}
+
+const DISH_CONSTANTS = Object.freeze({
+  DISH_LEN_STR: Object.freeze({
+    MIN_LENGTH: 1,
+    MAX_LENGTH: 500,
+  }),
+  DISH_LEN_NUMBER: Object.freeze({
+    MIN_LENGTH: 1,
+    MAX_LENGTH: 5,
+  }),
+});
+
+@Exclude()
+@Schema({
+  collection: 'dishes',
+})
+class Dish {
+  @Expose()
+  @Prop({
+    required: true,
+    minlength: DISH_CONSTANTS.DISH_LEN_STR.MIN_LENGTH,
+    maxlength: DISH_CONSTANTS.DISH_LEN_STR.MAX_LENGTH,
+  })
+  name: string;
+
+  @Expose()
+  @Prop({
+    type: [{ enum: MealType }],
+    minlength: DISH_CONSTANTS.DISH_LEN_STR.MIN_LENGTH,
+    maxlength: DISH_CONSTANTS.DISH_LEN_STR.MAX_LENGTH,
+  })
+  mealType: string[];
+
+  @Expose()
+  @Prop({
+    minlength: DISH_CONSTANTS.DISH_LEN_STR.MIN_LENGTH,
+    maxlength: DISH_CONSTANTS.DISH_LEN_STR.MAX_LENGTH,
+  })
+  description: string;
+
+  @Expose()
+  @Prop({
+    required: true,
+    minlength: DISH_CONSTANTS.DISH_LEN_NUMBER.MIN_LENGTH,
+    maxlength: DISH_CONSTANTS.DISH_LEN_NUMBER.MAX_LENGTH,
+  })
+  price: mongoose.Decimal128;
+
+  @Expose()
+  @Prop()
+  photo: Buffer;
+
+  @Expose()
+  @Prop({
+    type: [{ enum: DishTags }],
+    minlength: DISH_CONSTANTS.DISH_LEN_STR.MIN_LENGTH,
+    maxlength: DISH_CONSTANTS.DISH_LEN_STR.MAX_LENGTH,
+  })
+  tags: string[];
+
+  @Expose()
+  @Prop(
+    raw({
+      name: {
+        type: String,
+        minlength: DISH_CONSTANTS.DISH_LEN_STR.MIN_LENGTH,
+        maxlength: DISH_CONSTANTS.DISH_LEN_STR.MAX_LENGTH,
+      },
+      canBeExcluded: { type: Boolean, default: false },
+    }),
+  )
+  ingredients: object[];
+
+  @Expose()
+  @Prop({ type: [{ enum: Allergens }] })
+  allergens: string[];
+
+  @Expose()
+  @Prop({
+    min: DISH_CONSTANTS.DISH_LEN_NUMBER.MIN_LENGTH,
+    max: DISH_CONSTANTS.DISH_LEN_NUMBER.MAX_LENGTH,
+  })
+  portionWeight: number;
+
+  @Expose()
+  @Prop({
+    required: true,
+    min: DISH_CONSTANTS.DISH_LEN_NUMBER.MIN_LENGTH,
+    max: DISH_CONSTANTS.DISH_LEN_NUMBER.MAX_LENGTH,
+  })
+  calories: number;
+
+  @Expose()
+  @Prop(
+    raw({
+      per100g: {
+        type: Number,
+        required: true,
+        min: DISH_CONSTANTS.DISH_LEN_NUMBER.MIN_LENGTH,
+        max: DISH_CONSTANTS.DISH_LEN_NUMBER.MAX_LENGTH,
+      },
+      perPortion: Number,
+      min: DISH_CONSTANTS.DISH_LEN_NUMBER.MIN_LENGTH,
+      max: DISH_CONSTANTS.DISH_LEN_NUMBER.MAX_LENGTH,
+    }),
```

Comment:
To się powtarza trzykrotnie, może można by to jakoś wyciągnąć? 🤔

Reply:
byłoby spoko, ale na razie jeszcze nie wiem jak. Więc może jak czas starczy to się zrobi refaktor tego